sdks/python/apache_beam/testing/benchmarks/nexmark/nexmark_launcher.py

```diff
@@ -0,0 +1,199 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Nexmark launcher.
+
+The Nexmark suite is a series of queries (streaming pipelines) performed
+on a simulation of auction events. The launcher orchestrates the generation
+and parsing of streaming events and the running of queries.
+
+Model
+- Person: Author of an auction or a bid.
+- Auction: Item under auction.
+- Bid: A bid for an item under auction.
+
+Events
+ - Create Person
+ - Create Auction
+ - Create Bid
+
+Queries
+- Query0: Pass through (send and receive auction events).
+
+Usage
+- DirectRunner
+python nexmark_launcher.py \
+    --query/q <query number> \
+    --project <project id> \
+    --streaming \
+
+- DataflowRunner
+python nexmark_launcher.py \
+    --query/q <query number> \
+    --project <project id> \
+    --streaming \
+    --sdk_location <apache_beam tar.gz> \
+    --staging_location=gs://... \
+    --temp_location=gs://
+
+"""
+from __future__ import print_function
+
+import argparse
+import logging
+import sys
+
+import apache_beam as beam
+from apache_beam.options.pipeline_options import GoogleCloudOptions
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import SetupOptions
+from apache_beam.options.pipeline_options import StandardOptions
+from apache_beam.options.pipeline_options import TestOptions
+
+from nexmark_util import Command
+from queries import query0
+
+logging.basicConfig(level=logging.INFO,
+                    format='(%(threadName)-10s) %(message)s')
+
+# Reserved configuration to run these benchmark suite.
+PUBSUB_PROJECT = 'google.com:clouddfe'
```

Comment:
Please do not hardcode project name here. You can do this in the place where these tests will be executed. (Similar to: https://github.com/apache/beam/blob/master/sdks/python/scripts/run_postcommit.sh)

Reply:
As we talked offline, you can use a --project command line option instead of hardcoding here.